test/mpi/cxx/datatype/packsizex.cxx

```diff
@@ -81,30 +40,9 @@ int main(int argc, char *argv[])
 
     comm = MPI::COMM_WORLD;
 
-    /* TODO: parse input parameters using optarg */
-    if (argc < 3) {
-        cout << "Usage: " << argv[0] << "-type=[TYPE] -count=[COUNT]\n";
-        return 0;
-    } else {
-        for (i = 1; i < argc; i++) {
-            if (!strncmp(argv[i], "-type=", strlen("-type="))) {
-                j = 0;
-                while (strcmp(typelist[j].type_name.c_str(), "MPI_DATATYPE_NULL") &&
-                       strcmp(argv[i]+strlen("-type="), typelist[j].type_name.c_str())) {
-                    j++;
-                }
-
-                if (strcmp(typelist[j].type_name.c_str(), "MPI_DATATYPE_NULL")) {
-                    basic_type = typelist[j].type;
-                } else {
-                    cout << "Error: datatype not recognized\n";
-                    return 0;
-                }
-            } else if (!strncmp(argv[i], "-count=", strlen("-count="))) {
-                count = atoi(argv[i]+strlen("-count="));
-            }
-        }
-    }
+    err = MTestInitBasicSignatureX(argc, argv, &count, &basic_type);
```

Comment:
I assume 'X' stands for cxx and the reason for this name is to easily distinguish C++ version and C version. I like the idea, but I'd like to gather more feedback on this particular naming scheme. Is there any other use of 'X' like this?

Reply:
Yes X is for cxx. In `test/mpi/cxx` test names are suffixed by `x`. E.g., `test/mpi/cxx/attr/fkeyvaltypex.cxx`.